src/db.c

```diff
@@ -361,7 +361,14 @@ robj *dbUnshareStringValue(redisDb *db, robj *key, robj *o) {
         robj *decoded = getDecodedObject(o);
         o = createRawStringObject(decoded->ptr, sdslen(decoded->ptr));
         decrRefCount(decoded);
-        dbOverwrite(db,key,o);
+        dictEntry *de = dictFind(db->dict, key->ptr);
+        dictEntry auxentry = *de;
```

Comment:
maybe the assertion (with extra info) can still be useful?
```suggestion
        serverAssertWithInfo(NULL,key,de != NULL);
        dictEntry auxentry = *de;
```

Reply:
As the use case of `dbUnshareStringValue`, I think `de != NULL` is always `true`.

```
 o = lookupKeyWrite(db,key);
 if (checkType(c,o,OBJ_STRING)) return;
 o = dbUnshareStringValue(db,key,o);
```

Reply:
yes, but that's in another function, and who knows what other users will be using this function in the future...
doing a `*de` will crash too.. just thought that an assertions with key name may give an additional context for the crash log.
but i'm also ok with the current code.